lib/semantics/check-omp-structure.h

```diff
@@ -151,10 +152,13 @@ class OmpStructureChecker : public virtual BaseChecker {
   // collected information for END directive
   void ResetPartialContext(const parser::CharBlock &source) {
     CHECK(!ompContext_.empty());
+    GetContext().requiredClauses.IterateOverMembers(
+        [this](OmpClause c) { CheckRequired(c); });
```

Comment:
Not every construct would call `ResetPartialContext`. Besides, it's little weird to do the checks in the `Reset` function. How about in `Leave(const parser::OmpClauseList &)`?